shared/haml/user_header.haml

```diff
@@ -2,23 +2,33 @@
 :ruby
   require 'cdo/create_header'
   require 'cdo/language_dir'
-  limit_project_types = request.cookies["_limit_project_types_#{rack_env}"]
 
-  level = defined?(local_assigns) ?
-    local_assigns[:level] :
-    false
+  # Determine if we want to show the create menu
+  if local_assigns[:show_create_menu]
+    show_create_menu = local_assigns[:show_create_menu] == "true"
+  elsif local_assigns[:level]
+    # Only show the create menu on project levels (and not on levels in units)
+    show_create_menu = local_assigns[:level].try(:is_project_level)
+  else
+    # Show the create menu on non-level pages
+    show_create_menu = true
+  end
 
-  options = {
-    limit_project_types: limit_project_types,
-    project_type: params && params[:key]
-  }
-  create_drop_down_items = CreateHeader.get_create_dropdown_contents(options)
+  if show_create_menu
+    limit_project_types = request.cookies["_limit_project_types_#{rack_env}"]
 
-  # We want to show the Create button for all pages except when working on a
-  # course level. Specifically, for any page except non-project levels.
-  should_show_create_menu = !level || level.try(:is_project_level)
+    level = defined?(local_assigns) ?
```

Comment:
Do you need to keep this? Is `level` used anywhere?

Reply:
No, I don't need this anymore, nice catch!